stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+        unidecode_output = unidecode(token.text, errors='strict')
+        token.text = unidecode_output.translate(transl_table)
+        new_tokens.append(token)
+
+    return new_tokens
+
+
+def addSpacesToQuotes(text):
+    """ adds space before and after the quotes """
+
+    # adding space before and after quotes
+    regex_quotes = [r"(\S)(“)", r"(\S)(”)", r'(\S)(")', r"(\S)(“)", r"(\S)(‘)", r"(\S)(’)", r"(\S{1})(')(\s*)(\w{1,2})",
+                    r"(“)(\S)", r"(”)(\S)", r'(")(\S)', r"(“)(\S)", r"(‘)(\S)", r"(’)(\S)", r"(')(\w{1,2})"]
+
+    for idx, quote in enumerate(regex_quotes):
+        match = re.search(quote, text)
+        if match is not None:
+            text = re.sub(quote, lambda x: identifyMatchingGroups(x), text)
+
+    return text
+
+reportingVerbs = open('reportingVerbs.txt','r').read().split("\n")
+
+
+class TrigramMatching():
+    def __init__(self):
+        self.nlp = spacy.load("en_core_web_sm")
+        self.patterns = [
+            # Quote Verb Mention
+            [
+                {'POS': 'PUNCT'},
+                {'POS': 'VERB'},
+                {'POS': 'NOUN'}
+            ],
+            [
+                {'POS': 'PUNCT'},
+                {'POS': 'VERB'},
+                {'POS': 'PROPN'}
+            ],
+            # Quote Mention Verb
+            [
+                {'POS': 'PUNCT'},
+                {'POS': 'NOUN'},
+                {'POS': 'VERB'}
+
+            ],
+            [
+                {'POS': 'PUNCT'},
+                {'POS': 'PRON'},
+                {'POS': 'VERB'}
+            ],
+            # Mention Verb Quote
+            [
+                {'POS': 'NOUN'},
+                {'POS': 'VERB'},
+                {'POS': 'PUNCT'}
+
+            ],
+            [
+                {'POS': 'PRON'},
+                {'POS': 'VERB'},
+                {'POS': 'PUNCT'}
+            ],
+            # Verb Mention Quote
+            [
+                {'POS': 'VERB'},
+                {'POS': 'NOUN'},
+                {'POS': 'PUNCT'}
+            ],
+            [
+                {'POS': 'VERB'},
+                {'POS': 'PRON'},
+                {'POS': 'PUNCT'}
+            ],
+
+        ]
+
+    def getMentionFromPattern(self, pattern, tokens):
+        noun = None
+        tokens = tokens.split()
+        for i, kvPair in enumerate(pattern):
+            if kvPair['POS'] == 'NOUN' or kvPair['POS'] == 'PRON':
+                noun = tokens[i]
+                break
+        return noun
+
+    def getMention(self, text):
+        for i, pattern in enumerate(self.patterns):
+            matcher = Matcher(self.nlp.vocab)
+            matcher.add(f"Trigram_{i}", [pattern])
+
+            doc = self.nlp(text)
+
+            matches = matcher(doc)
+            for match_id, start, end in matches:
+                string_id = self.nlp.vocab.strings[match_id]
+                span = doc[start:end]
+                return self.getMentionFromPattern(pattern, span.text)
+
+patterns = [
```

Comment:
Avoid using global variables, we do not want to pollute the namespace when importing this file. Add a function that returns the pattern